### PR TITLE
Closes #2034 use functions to standardize roxygen text

### LINKED ISSUE #2034: Add roxygen2 template for common arg names/descriptions
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

http://cran.nexr.com/web/packages/roxygen2/vignettes/rd.html

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
For context: 

image

File: R/call_derivation.R [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #' Call a single derivation multiple times with some parameters/arguments being fixed across
 #' iterations and others varying.
 #'
-#' @param dataset The input dataset
+#' @param dataset Input dataset
```
Comment (zdz2101) [inline R/call_derivation.R:6]: Many of the "labels" are `Input dataset`, task should include modifying all non-conforming ones to this

File: R/create_single_dose_dataset.R [hunk 2]
```diff
@@ -187,8 +187,7 @@ dose_freq_lookup <- tribble(
 #'
 #' @param dataset Input dataset
 #'
-#'   The columns specified by `dose_freq`, `start_date` and the `end_date`
-#'   parameters are expected.
+#'   `r roxygen_param_dataset(expected_vars_args = c("dose_freq", "start_date", "end_date"))`
```
Comment (zdz2101) [inline R/create_single_dose_dataset.R:190]: @bundfussr Is this what you had in mind? Easier to see in action in the `.Rd` (screenshotted here) 

image

Comment (bundfussr) [reply R/create_single_dose_dataset.R:190]: > Is this what you had in mind?

Yes.

I would create the label ("Input dataset") by the function as well. Then it is consistent.

Maybe we could even create `@param dataset` by the function. But I am not sure if this works.

File: R/derive_adeg_params.R [hunk 3]
```diff
@@ -6,10 +6,9 @@
 #'
 #' **Note:** This is a wrapper function for the more generic `derive_param_computed()`.
 #'
-#' @param dataset Input dataset
-#'
-#'   The variables specified by the `by_vars` and the `unit_var` parameter,
-#'   `PARAMCD`, and `AVAL` are expected.
+#' @param dataset
+#'   `r roxygen_param_dataset(expected_vars = c("by_vars", "get_unit_expr"))`
+#'   `PARAMCD`, and `AVAL` are expected as well.
```
Comment (zdz2101) [inline R/derive_adeg_params.R:11]: Other examples of explicit column names from `derive_var_atoxgr()`: 

Whats best way to handle?
image

Comment (zdz2101) [reply R/derive_adeg_params.R:11]: @pharmaverse/admiral Do you all have thoughts on how to handle phrasing of explicitly named variables vs the ones we plug into function arguments?

File: R/call_derivation.R [hunk 4]
```diff
@@ -3,7 +3,7 @@
 #' Call a single derivation multiple times with some parameters/arguments being fixed across
 #' iterations and others varying.
 #'
-#' @param dataset The input dataset
+#' @param dataset  `r roxygen_param_dataset()`
```
Comment (zdz2101) [inline R/call_derivation.R:6]: @pharmaverse/admiral Is *just* "Input dataset" the appropriate description, or should all entries mention something about expected variables?

Comment (bundfussr) [reply R/call_derivation.R:6]: I think in this case just "Input dataset" is OK.

**Comment by zdz2101** [comment]
To summarize what was done:

the `@param dataset` label and description is now programatically generated:

- if only the label is needed, inline code `roxygen_param_dataset()` is called to create the "Input dataset" label which is now standardized across all functions
- functions are now aligned with their corresponding `assert_data_frame(dataset,. required_vars = ... )` call in the function body where essentially the `expected_vars` argument matches the `required_vars` argument
- some functions have explicit calls for specific column names like `PARAMCD` or `AVAL` , documentation primarily remained unchanged, but the sentence explicitly writing this may have been standardized to:
   - ` #'  PARAMCD, and AVAL are expected as well.`
- `# nolint` doesn't work in roxygen, so the programmatic standardized description of `derive_var_trtemfl()` and `derive_var_ontrtfl()` was **not** done

**Comment by bms63** [comment]
@zdz2101 can you resolve this conflict please - I think this is ready to go after that right?

**Comment by zdz2101** [comment]
> @zdz2101 can you resolve this conflict please - I think this is ready to go after that right?

Yea should be good to go, hopefully no man page .Rd fix necessary, it passed through my local `devtools::document()`

**Comment by bms63** [comment]
ooohh - we should have it go to `main`

**Comment by zdz2101** [comment]
> ooohh - we should have it go to `main`

oh my, gimme a moment

**Review by bms63 — APPROVED** [review]
Well here goes - probably check the dev website after the branch rebuilds

**Comment by zdz2101** [comment]
@bms63 I don't think it "auto" re-renders without that upversioning of the `DESCRIPTION` file? But are we still going to use that action that does it for us that we talked about on wednesday?

**Pull request merged**